Src/IronPython/Runtime/PythonFileManager.cs

```diff
@@ -28,10 +29,19 @@ internal sealed class StreamBox {
         private int _id = -1;
         private Stream _readStream;
         private Stream _writeStream;
+        private readonly bool _writeNeedsFlush;
 
         public StreamBox(Stream readStream, Stream writeStream) {
             _readStream = readStream;
             _writeStream = writeStream;
+
+            // a hack but it does improve perfomance a lot if flushing is not needed
+            _writeNeedsFlush = writeStream switch {
+                FileStream => true,         // FileStream is buffered by default, used on Windows and mostly Mono
+                PosixFileStream => false,   // PosixFileStream is unbuffered, used on .NET/Posix
+                UnixStream => false,        // UnixStream is unbuffered, used sometimes on Mono
```

Comment:
I guess this is causing the test failures since it'll try to load the `Mono.Posix` assembly (which isn't packaged along with Windows builds).

Reply:
Indeed, it makes sense... The failure on Linux is a bit more interesting as it is failing when dealing with a file opened in the "a" mode for simultaneous logging.